lib/esefilereader.cpp

```diff
@@ -0,0 +1,105 @@
+/* ESExtractor
+ * Copyright (C) 2023 Igalia, S.L.
+ *     Author: Stephane Cerveau <scerveau@igalia.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License.  You
+ * may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+#include "esefilereader.h"
+
+ESEFileReader::ESEFileReader (const char *fileName)
+{
+  if (fileName)
+    m_fileName = fileName;
+  reset ();
+}
+
+ESEFileReader::~ESEFileReader ()
+{
+}
+
+bool
+ESEFileReader::prepare ()
+{
+  if (m_file.is_open ())
+    return true;
+
+  if (m_fileName.empty ())
+    return false;
+
+  m_file = std::ifstream (m_fileName, std::ios::binary | std::ios::ate);
+  DBG ("The file %s is now %s", m_fileName.c_str (), m_file.is_open () ? "open" : "closed");
+  if (m_file.is_open ()) {
+    m_streamSize = m_file.tellg ();
```

Comment:
what's the purpose of setting m_streamSize to zero?

would be something like 

```
m_file.seekg(0, end);
m_streamSize = m_file.tellg ();
m_file.seekg(0);
```

Reply:
Valgrind is complaining that this value can be used unset.

I rewrote a little bit in `prepare` the way to initialize it.